build/cmake/lib/webview_chromium/CMakeLists.txt

```diff
@@ -62,8 +70,149 @@ if(NOT CEF_ROOT)
     if(result)
         message(FATAL_ERROR "Build step for cef failed: ${result}")
     endif()
+endfunction()
+
+if(NOT CEF_ROOT)
+    include(cef_version_info.cmake)
+
+    # Note that CMAKE_OSX_ARCHITECTURES is always defined under macOS, but is
+    # empty if not explicitly specified, so do _not_ use if(DEFINED ...) here.
+    if(CMAKE_OSX_ARCHITECTURES)
+        foreach(osx_arch IN LISTS CMAKE_OSX_ARCHITECTURES)
+            # Determine the matching CEF platform name.
+            if(${osx_arch} STREQUAL "arm64")
+                set(CEF_PLATFORM "macosarm64")
+            elseif(${osx_arch} STREQUAL "x86_64")
+                set(CEF_PLATFORM "macosx64")
+            else()
+                message(FATAL_ERROR "Building CEF for macOS architecture ${osx_arch} is not supported")
+            endif()
+
+            if(NOT first_cef_platform)
+                # Download/unpack CEF files in the root directory and remember
+                # that we did it by setting this variable.
+                set(first_cef_platform ${CEF_PLATFORM})
+            else()
+                list(APPEND other_cef_platforms ${CEF_PLATFORM})
+
+                # Use different subdirectories for the other architectures.
+                set(CEF_DOWNLOAD_DIR "${CEF_DOWNLOAD_ROOT}/${CEF_PLATFORM}")
+                set(CEF_SOURCE_DIR "${CEF_SOURCE_ROOT}/${CEF_PLATFORM}")
+                set(CEF_BUILD_DIR "${CEF_BUILD_ROOT}/${CEF_PLATFORM}")
+            endif()
+
+            wx_download_cef(${CEF_PLATFORM})
+        endforeach()
+
+        # Now lipo all the binaries together unless this is a degenerate case
+        # in which CMAKE_OSX_ARCHITECTURES contains only one element.
+        if(other_cef_platforms)
+            set(cef_binaries
+                "cef_sandbox.a"
+                "Chromium Embedded Framework.framework/Chromium Embedded Framework"
+                "Chromium Embedded Framework.framework/Libraries/libEGL.dylib"
+                "Chromium Embedded Framework.framework/Libraries/libGLESv2.dylib"
+                "Chromium Embedded Framework.framework/Libraries/libvk_swiftshader.dylib"
+            )
+
+            foreach(cef_bin_file IN LISTS cef_binaries)
+                set(lipo_command
+                    lipo -create "Release/${cef_bin_file}"
+                )
+
+                foreach(cef_platform IN LISTS other_cef_platforms)
+                    list(APPEND lipo_command "${cef_platform}/Release/${cef_bin_file}")
+                endforeach()
+
+                list(APPEND lipo_command "-output")
+                list(APPEND lipo_command "Release/${cef_bin_file}.tmp")
```

Comment:
Why is the `.tmp` file and rename needed?

Reply:
I don't think lipo can modify files in place, so this creates a temporary file and then renames it to the original one, how else can we do it?

BTW, one problem with this is that once lipo has been run, running it again will fail because it will complain about the other arch being already present in the first source file (which will be a fat one). I don't know if it's worth catering for this...